deployment/clusterObjectModel/forward_compatibility.py

```diff
@@ -65,11 +65,19 @@ def service_configuration_convert(service_configuration):
             service_configuration["hadoop-resource-manager"]["yarn_exporter_port"] = service_configuration["prometheus"]["yarn_exporter_port"]
         if "prometheus-port" in service_configuration["prometheus"] and "port" not in service_configuration["prometheus"]:
             service_configuration["prometheus"]["port"] = service_configuration["prometheus"]["prometheus-port"]
+        if "alerting" in service_configuration["prometheus"]:
+            service_configuration["alert-manager"] = service_configuration["prometheus"]["alerting"]
```

Comment:
does alert-manager always exist?

Reply:
Depend on whether ```alerting``` exits or not.

Reply:
Changed

Reply:
With so many ifs, I think maybe you should implement a function that can do this.

Maybe `transform(old_model, "prometheus.alert.alert_manager_port", new_model, "alert-manager.receiver")` which means copy one field from one object to another field of another object, and handle this no-exist field problem in the transform function. This way it's more easy to debug and find bugs.